pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -421,6 +422,14 @@ func (c *Controller) injectPacket(tf *opsv1alpha1.Traceflow) error {
 		idICMP = uint16(tf.Spec.Packet.TransportHeader.ICMP.ID)
 		sequenceICMP = uint16(tf.Spec.Packet.TransportHeader.ICMP.Sequence)
 	}
+
+	var packetOutIcmpEchoRequestType icmpType
+	if isIPv6 {
+		packetOutIcmpEchoRequestType = icmpv6EchoRequestType
+	} else {
+		packetOutIcmpEchoRequestType = icmpEchoRequestType
+	}
+
 	return c.ofClient.SendTraceflowPacket(
```

Comment:
It is not introduced by this PR, but would you change SendTraceflowPacket to check IP protocol against the protocol constants, instead of integer values?

Reply:
For SendTraceflowPacket function, I prefer to use protocol uint8 as input since it is a low level function, and this is verified inside this function.

Reply:
But it is hard to understand and maintain the code. IMO, we should use constants for such cases.

Reply:
Would you make the change, or maybe you would consider a separate PR?

Reply:
I think we should discuss about this with @wenyingd in a separate PR.
User defines protocol in integer in Traceflow CRD, and passed it to this function, and we verify this property with predefined constant in libopenflow. I prefer keep the protocol as integer since this is a common value in integer type.

Reply:
It might be my mistake or the code has been updated already. I remember I saw SendTraceflowPacket() is comparing IP protocol with integer values (e.g. 6 for TCP), but I now saw the latest code is already using constants. If I was not talking about the same thing as you and caused confusion, I would applogize.

But I would think a constant should be defined for any constant value referred in code, which is much readable and easier to maintain. You do not want any reader to check any occurrence of an integer value even maybe in this case there might be only one occurrence for now? It is irrelevant user input is integer value or not.